modules/super-admin/components/product-listing/ProductListingTable.tsx

```diff
@@ -9,105 +7,77 @@ import Button from '@ui/Button';
-      vendor: 'Mark Essien',
-      id: 1236,
-      dateAdded: '08-03-23',
-      status: 'Active',
-    },
-    {
-      productName: 'Ethical Hacking',
-      vendor: 'Shobande Abraham',
-      id: 1237,
-      dateAdded: '08-04-23',
-      status: 'Active',
-    },
-    {
-      productName: 'Product Management',
-      vendor: 'John Paul',
-      id: 1238,
-      dateAdded: '08-05-23',
-      status: 'Active',
-    },
-    {
-      productName: 'Digital Marketing',
-      vendor: 'Bakare Femi',
-      id: 1239,
-      dateAdded: '08-06-23',
-      status: 'Active',
-    },
-    {
-      productName: 'Fullstack Development',
-      vendor: 'Gustavo Silas',
-      id: 1233,
-      dateAdded: '08-07-23',
-      status: 'Active',
-    },
-  ];
-  const [searchVal, setSearchVal] = useState('');
-  const [filteredProducts, setFilteredProducts] = useState(data);
+interface ExtractedProductInfo {
+  product_id: number;
+  product_name: string;
+  status: string;
+  vendor_name: string;
+  date_added: string;
+}
+
+const ProductListingTable: React.FC = () => {
+  // State to hold the fetched data from the endpoint
+  const [data, setData] = useState<ProductInfo[]>([]);
+  // State for search input value
+  const [searchVal, setSearchVal] = useState<string>('');
+  // State to hold the filtered products based on search and sorting
+  const [filteredProducts, setFilteredProducts] = useState<ExtractedProductInfo[]>([]);
+
+  useEffect(() => {
+    // Fetch data from the endpoint when the component mounts
+    const fetchData = async () => {
+      try {
+        const response = await fetch('https://test-super-admin.onrender.com/api/product/get-all-products/');
+        const result = await response.json();
+
+        // Ensure result is an array
+        const resultArray = Array.isArray(result) ? result : [];
+
+        setData(resultArray);
+
+        // Extracting specific fields: product_id, product_name, status, vendor_name
+        const extractedData: ExtractedProductInfo[] = resultArray.map((product: ProductInfo) => ({
+          product_id: product.product_id,
+          product_name: product.product_name,
+          status: product.status,
+          vendor_name: product.vendor_name,
+          date_added: product.date_added,
+        }));
+
+        setFilteredProducts(extractedData);
+
+        // Log the extracted data
+        console.log('Extracted Data:', extractedData);
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, []);
 
   const handleSearch = (searchText: string) => {
-    const filteredProduct: Array<{
-      productName: string;
-      vendor: string;
-      id: number;
-      dateAdded: string;
-      status: string;
-    }> = data.filter((product) => product.productName.toLowerCase().includes(searchText.toLowerCase()));
+    // Filter products based on the search text
+    const filteredProduct = data.filter((product) =>
```

Comment:
What if I want to search for products belonging to a particular vendor, will this filter logic quell my thirst?